tests/jobscript/18-env-task-dependencies.t

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2018 NIWA
+# 
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Test job script sets dependencies in evironment.
+. "$(dirname "${0}")/test_header"
+set_test_number 3
+
+install_suite "${TEST_NAME_BASE}" "${TEST_NAME_BASE}"
+run_ok "${TEST_NAME_BASE}-validate" cylc validate "${SUITE_NAME}"
+run_ok "${TEST_NAME_BASE}-run" \
+    cylc run "${SUITE_NAME}" --reference-test --debug --no-detach
+grep_ok 'CYLC_TASK_DEPENDENCIES="bar\.1 baz\.1"' \
+    "${SUITE_RUN_DIR}/log/job/1/foo/01/job"
+
```

Comment:
Add a grep for null `CYLC_TASK_DEPENDENCIES` in `qux/01/job`?

Reply:
Done